src/lib/forms/rjsf/widgets/RJSFESSelector.js

```diff
@@ -0,0 +1,164 @@
+import { withCancel } from '@api/utils';
+import _debounce from 'lodash/debounce';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { Form } from 'semantic-ui-react';
+
+/**
+ * Generic React JSONSchema Form widget to search and select hits
+ * on Elasticsearch.
+ * It is meant to be wrapped in specialized component and not to
+ * be directly used.
+ */
+export class RJSFESSelector extends Component {
+  constructor(props) {
+    super(props);
+
+    const { value } = this.props;
+    this.state = {
+      isLoading: false,
+      value: value,
+      error: null,
+      options: [],
+    };
+  }
+
+  async componentDidMount() {
+    const { value } = this.props;
+    if (value) {
+      this.fetchValue(value);
+    }
+  }
+
+  componentWillUnmount() {
+    this.cancellableFetchData && this.cancellableFetchData.cancel();
+  }
+
+  async fetchValue(value) {
+    const { options } = this.props;
+    const { apiGetByValue, apiGetByValueResponseSerializer } = options;
+    this.cancellableFetchData = withCancel(apiGetByValue(value));
+    try {
+      this.setState({
+        isLoading: true,
+      });
+
+      const response = await this.cancellableFetchData.promise;
+      const singleOption = apiGetByValueResponseSerializer(response.data);
+
+      this.setState({
+        isLoading: false,
+        value: singleOption.value,
+        options: [singleOption],
+      });
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        this.setState({
+          isLoading: false,
+          error: 'Error loading results.',
+        });
+      }
+    }
+  }
+
+  handleSearchChange = async (e, { searchQuery }) => {
+    const { options, required } = this.props;
+    const { apiQuery, apiQueryResponseSerializer } = options;
+    if (searchQuery === '') {
+      return;
+    }
+
+    this.cancellableFetchData = withCancel(apiQuery(searchQuery));
+    try {
+      this.setState({
+        isLoading: true,
+      });
+
+      const response = await this.cancellableFetchData.promise;
+      const results = response.data.hits.map((hit) =>
+        apiQueryResponseSerializer(hit)
+      );
+
+      if (!required) {
+        // add an empty element at the beginning so that if it is not required, you can select nothing
+        results.unshift({ key: '', value: undefined, text: '' });
```

Comment:
```suggestion
        results.unshift({ key: '', value: undefined, text: '-' });
```
Comment: IMHO if it's empty it's harder to recognise you can actually select it, with the hyphen maybe it's more user-friendly. WDYT?